core/include/array/array_sorted_read_state.h

```diff
@@ -228,7 +228,7 @@ class ArraySortedReadState {
   AIO_Request aio_request_[2];
 
   /** The status of the AIO requests.*/
-  int aio_status_[2];
+  tiledb_aio_state_t aio_status_[2];
```

Comment:
Perhaps we should be consistent here too regarding `status` and `state`. I used `status` here in `aio_status_` because this captures error codes. I am not sure what the best name here should be. I am ok with either.